src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,452 @@
+                await this.ExecuteCommandAsync("tar", $"-zxvf {this.PackageName}.tar.gz", this.PlatformSpecifics.PackagesDirectory, telemetryContext, cancellationToken);
+                state.HPLInitialized = true;
+
+            }
+
+            this.HPLDirectory = this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, $"hpl-{this.HPLVersion}");
+            await this.stateManager.SaveStateAsync<HPLinpackState>($"{nameof(HPLinpackState)}", state, cancellationToken);
+
+            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, "Make.Linux_GCC"))
+                .ConfigureAwait(false);
+
+            await this.DeleteFileAsync(this.PlatformSpecifics.Combine(this.HPLDirectory, "setup", "Make.Linux_GCC"))
+                .ConfigureAwait(false);
+
+            await this.ExecuteElevatedCommandAsync("bash", "-c \"source make_generic\"", this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup"), telemetryContext, cancellationToken);
+
+            await this.ConfigureMakeFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            await this.ExecuteCommandAsync("ln", $"-s {this.PlatformSpecifics.Combine(this.HPLDirectory, $"setup", $"Make.Linux_GCC ")} Make.Linux_GCC", this.HPLDirectory, telemetryContext, cancellationToken);
+
+        }
+
+        /// <summary>
+        /// Executes the HPL workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            DateTime startTime = DateTime.UtcNow;
+            await this.ExecuteCommandAsync("make", $"arch=Linux_GCC", this.HPLDirectory, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.SetParameters();
+            await this.ConfigureHPLdatFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            string results;
+            if (this.CpuArchitecture == Architecture.X64)
+            {
+                results = await this.ExecuteElevatedCommandAsync("runuser", $"-u azureuser -- mpirun --use-hwthread-cpus -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+            else
+            {
+                results = await this.ExecuteElevatedCommandAsync("runuser", $"-u azureuser -- mpirun -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+
+            DateTime endTime = DateTime.UtcNow;
+            this.LogMetrics(results, startTime, endTime, telemetryContext, cancellationToken);
+        }
+
+        private void SetParameters()
+        {
+            if (this.N == null)
```

Comment:
This can be set in the property at the top passing in a default value.

this.Parameters.GetValue<int>(nameof(.....), Environment.ProcessorCount * 10000)